ui/partials/main/thPinboardPanel.html

```diff
@@ -56,6 +56,15 @@
                ng-click="allowKeys()"
                placeholder="click to add comment"
                blur-this></input>
+        <div ng-if="classification.failure_classification_id === 2">
+          <div class="input-group-btn">
+              <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" style="padding: 2px 12px;">
```

Comment:
I was thinking that perhaps you could use a ``<select>`` element instead of a ``button`` and ``dropdown`` so that it looks visually consistent with the ``<select>`` above with the ``classificaiton_id``.  Perhaps something like this?

    <div ng-if="classification.failure_classification_id === 2">
      <select id="recent-choice">
        <option value="0" selected>Choose a recent commit</option>
        <option ng-repeat="tip in originalTipList | limitTo:20"
          ng-click="classification.text=tip.revision">{{tip.revision | limitTo: 12}} {{tip.author}}
        </option>
      </select>
    </div>

And this means you wouldn't need some of the extra functions in ``pinboard.js``.  Only the ones to populate the ``originalTipList``.